### PR TITLE
Cleanup monitoring secrets when purpose is switched to testing

### LINKED ISSUE #5826: Delete `<shoot-name>.monitoring` secret from project namespace when purpose is switched to `testing`
**How to categorize this issue?**

/area monitoring
/kind bug

**What happened**:
When an existing `Shoot` with a `purpose!=testing` is switched to `purpose=testing` then the monitoring stack is deleted: https://github.com/gardener/gardener/blob/61e391ecb37013c8fa0637136bb03ba9f4ddac91/pkg/operation/botanist/monitoring.go#L452-L454

However, this does not delete the `<shoot-name>.monitoring` secret from the project namespace 
This secret is originally synced to the project namespace in the `DeploySeedGrafana` function: https://github.com/gardener/gardener/blob/61e391ecb37013c8fa0637136bb03ba9f4ddac91/pkg/operation/botanist/monitoring.go#L519-L525
We would need similar cleanup code like here: https://github.com/gardener/gardener/blob/61e391ecb37013c8fa0637136bb03ba9f4ddac91/pkg/operation/botanist/kubeapiserver.go#L480-L483

**What you expected to happen**:
The `<shoot-name>.monitoring` secret should disappear if no monitoring stack is deployed anymore.

**How to reproduce it (as minimally and precisely as possible)**:
1. Create a shoot with `purpose=evaluation`.
2. See the `<shoot-name>.monitoring` secret appear in the project namespace
3. Update the shoot to `purpose=testing`.
4. Still see the `<shoot-name>.monitoring` secret existing in the project namespace

### PR DESCRIPTION
**Pull request by shafeeqes**
**How to categorize this PR?**

/area monitoring
/kind bug

**What this PR does / why we need it**:
The `<shoot-name>.monitoring` secret in the project namespace is now properly cleaned up when purpose is switched to testing (ref #5608)

**Which issue(s) this PR fixes**:
Fixes #5826 

**Special notes for your reviewer**:

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Review by rfranzke — COMMENTED** [review]
Looks good! Can we extend the unit test (https://github.com/gardener/gardener/blob/39ef719a614036ef8979d15141309e19c8cf8a35/pkg/operation/botanist/monitoring_test.go#L225) with this?

File: pkg/operation/botanist/monitoring_test.go [hunk 1]
```diff
@@ -280,5 +286,21 @@ var _ = Describe("Monitoring", func() {
...
+
+			Expect(botanist.DeploySeedGrafana(ctx)).To(Succeed())
+			Expect(fakeGardenClient.Get(ctx, kutil.Key(projectNamespace, shootName+".monitoring"), &corev1.Secret{})).To(Succeed())
+
+			botanist.Shoot.Purpose = shootPurposeTesting
+			Expect(botanist.DeploySeedGrafana(ctx)).To(Succeed())
+
+			Eventually(func(g Gomega) error {
+				return fakeGardenClient.Get(ctx, kutil.Key(projectNamespace, shootName+".monitoring"), &corev1.Secret{})
+			}).Should(BeNotFoundError())
```
Comment (rfranzke) [inline pkg/operation/botanist/monitoring_test.go]: Why `Eventually`? I think you can assert right away.

File: pkg/operation/botanist/monitoring_test.go [hunk 2]
```diff
@@ -280,5 +286,21 @@ var _ = Describe("Monitoring", func() {
...
+
+			Expect(fakeGardenClient.Get(ctx, kutil.Key(projectNamespace, shootName+".monitoring"), &corev1.Secret{})).To(BeNotFoundError())
+
+			Expect(botanist.DeploySeedGrafana(ctx)).To(Succeed())
+			Expect(fakeGardenClient.Get(ctx, kutil.Key(projectNamespace, shootName+".monitoring"), &corev1.Secret{})).To(Succeed())
+			Expect(*botanist.Shoot.GetInfo().Spec.Purpose == shootPurposeEvaluation).To(BeTrue())
+
+			botanist.Shoot.Purpose = shootPurposeTesting
+			Expect(botanist.DeploySeedGrafana(ctx)).To(Succeed())
+			Expect(botanist.Shoot.Purpose == shootPurposeTesting).To(BeTrue())
```
Comment (rfranzke) [inline pkg/operation/botanist/monitoring_test.go]: Drop this line, it doesn't yield any benefit, does it?

**Review by rfranzke — COMMENTED** [review]
one last nit :)

File: pkg/operation/botanist/monitoring_test.go (hunk 2 above)
Comment (shafeeqes) [reply pkg/operation/botanist/monitoring_test.go]: ohh.. sorry. I forgot to remove it.

**Review by rfranzke — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**